diff -Naur calamares-3.2.7/src/modules/users/UsersPage.cpp calamares-3.2.7-p/src/modules/users/UsersPage.cpp
--- calamares-3.2.7/src/modules/users/UsersPage.cpp	2019-04-27 20:12:03.000000000 +0200
+++ calamares-3.2.7-p/src/modules/users/UsersPage.cpp	2019-05-05 20:31:21.081178113 +0200
@@ -35,6 +35,7 @@
 #include "Settings.h"
 
 #include <QBoxLayout>
+#include <QFile>
 #include <QLabel>
 #include <QLineEdit>
 #include <QRegExp>
@@ -272,7 +273,27 @@
     {
         if ( !cleanParts.isEmpty() && !cleanParts.first().isEmpty() )
         {
-            QString hostnameSuggestion = QString( "%1-pc" ).arg( cleanParts.first() );
+
+            QString dmiProductName;
+            QString hostnameSuggestion;
+            // yes validateHostnameText() but these files sometimes have a lot crap in it
+            QRegExp dmirx( "[^a-zA-Z0-9]", Qt::CaseInsensitive );
+            QFile dmiFile( QStringLiteral( "/sys/devices/virtual/dmi/id/product_name" ) );
+
+            if ( dmiFile.exists() &&
+                 dmiFile.open(QIODevice::ReadOnly))
+            {
+                dmiProductName = QString::fromLocal8Bit( dmiFile.readAll().simplified().data() )
+                                 .toLower().replace(dmirx, " ").remove(' ');
+            }
+            if ( !dmiProductName.isEmpty() )
+            {
+                hostnameSuggestion = QString( "%1-%2" ).arg( cleanParts.first() ).arg( dmiProductName );
+            }
+            else
+            {
+               hostnameSuggestion = QString( "%1-pc" ).arg( cleanParts.first() );
+            }
             if ( HOSTNAME_RX.indexIn( hostnameSuggestion ) != -1 )
             {
                 ui->textBoxHostname->setText( hostnameSuggestion );
-- 
2.19.1

