diff -Naur calamares-3.2.7/src/modules/services-systemd/main.py calamares-3.2.7-p/src/modules/services-systemd/main.py
--- calamares-3.2.7/src/modules/services-systemd/main.py	2019-04-27 20:12:03.000000000 +0200
+++ calamares-3.2.7-p/src/modules/services-systemd/main.py	2019-05-05 20:29:14.883410685 +0200
@@ -72,12 +72,22 @@
                     description = _("Cannot enable systemd service <code>{name!s}</code>.")
                 elif command == "enable" and suffix == ".target":
                     description = _("Cannot enable systemd target <code>{name!s}</code>.")
+                elif command == "enable" and suffix == ".socket":
+                    description = _("Cannot enable systemd socket <code>{name!s}</code>.")
+                elif command == "enable" and suffix == ".timer":
+                    description = _("Cannot enable systemd timer <code>{name!s}</code>.")
                 elif command == "disable" and suffix == ".service":
-                    description = _("Cannot enable systemd service <code>{name!s}</code>.")
+                    description = _("Cannot disable systemd service <code>{name!s}</code>.")
                 elif command == "disable" and suffix == ".target":
                     description = _("Cannot disable systemd target <code>{name!s}</code>.")
+                elif command == "disable" and suffix == ".socket":
+                    description = _("Cannot disable systemd socket <code>{name!s}</code>.")
+                elif command == "disable" and suffix == ".timer":
+                    description = _("Cannot disable systemd timer <code>{name!s}</code>.")
                 elif command == "mask":
                     description = _("Cannot mask systemd unit <code>{name!s}</code>.")
+                elif command == "unmask":
+                    description = _("Cannot unmask systemd unit <code>{name!s}</code>.")
                 else:
                     description = _("Unknown systemd commands <code>{command!s}</code> and <code>{suffix!s}</code> for unit {name!s}.")
 
@@ -106,10 +116,26 @@
     if r is not None:
         return r
 
+    r = systemctl(cfg.get("sockets", []), "enable", ".socket")
+    if r is not None:
+        return r
+
+    r = systemctl(cfg.get("timers", []), "enable", ".timer")
+    if r is not None:
+        return r
+
     r = systemctl(cfg.get("disable", []), "disable", ".service")
     if r is not None:
         return r
 
+    r = systemctl(cfg.get("disable-sockets", []), "disable", ".socket")
+    if r is not None:
+        return r
+
+    r = systemctl(cfg.get("disable-timers", []), "disable", ".timer")
+    if r is not None:
+        return r
+
     r = systemctl(cfg.get("mask", []), "mask", "")
     if r is not None:
         return r
 
+    r = systemctl(cfg.get("unmask", []), "unmask", "")
+    if r is not None:
+        return r
 
     # This could have just been return r
     return None
diff --git a/src/modules/services-systemd/services-systemd.conf b/src/modules/services-systemd/services-systemd.conf
index 6ff1409bf..e5897e5b6 100644
--- a/src/modules/services-systemd/services-systemd.conf
+++ b/src/modules/services-systemd/services-systemd.conf
@@ -43,6 +43,16 @@
 #   - name: "graphical"
 #     mandatory: true
 #
+# # Enables <name>.socket
+# sockets:
+#   - name: "some-socket"
+#     mandatory: false
+#
+# # Enables <name>.timer
+# timers:
+#   - name: "some-timer"
+#     mandatory: true
+#
 # # Disables <name>.service
 # disable:
 #   - name: "pacman-init"
@@ -53,6 +63,16 @@
 # disable-targets:
 #   - graphical
 #
+# # Disables <name>.socket
+# #     .. this shows how to use just the name
+# disable-sockets:
+#   - some-socket
+#
+# # Disables <name>.timer
+# #     .. this shows how to use just the name
+# disable-timers:
+#   - some-timer
+#
 # # Masks (stronger version of disable). This section
 # # is unusual because you **must** include the suffix
 # # (e.g. ".service") as part of the name, so, e.g. to mask
@@ -61,10 +81,19 @@
 # mask:
 #  - name: "NetworkManager.service"
 #  - mandatory: true
-
+#
+# unmask:
+#  - name: "some.socket"
+#  - mandatory: true
+#
 # By default, no changes are made.
 services: []
+sockets: []
+timers: []
 targets: []
 disable: []
 disable-targets: []
+disable-sockets: []
+disable-timers: []
 mask: []
+unmask: []
-- 
2.20.1

