diff -up calamares-3.2.12/src/modules/users/UsersPage.cpp.3~ calamares-3.2.12/src/modules/users/UsersPage.cpp
--- calamares-3.2.12/src/modules/users/UsersPage.cpp.3~	2019-08-07 12:58:22.000000000 +0200
+++ calamares-3.2.12/src/modules/users/UsersPage.cpp	2019-08-22 15:31:35.548605727 +0200
@@ -40,6 +40,7 @@
 #include "utils/String.h"
 
 #include <QBoxLayout>
+#include <QFile>
 #include <QLabel>
 #include <QLineEdit>
 #include <QRegExp>
@@ -277,7 +278,27 @@ UsersPage::fillSuggestions()
     {
         if ( !cleanParts.isEmpty() && !cleanParts.first().isEmpty() )
         {
-            QString hostnameSuggestion = QString( "%1-pc" ).arg( cleanParts.first() );
+
+            QString dmiProductName;
+            QString hostnameSuggestion;
+            // yes validateHostnameText() but these files sometimes have a lot crap in it
+            QRegExp dmirx( "[^a-zA-Z0-9]", Qt::CaseInsensitive );
+            QFile dmiFile( QStringLiteral( "/sys/devices/virtual/dmi/id/product_name" ) );
+
+            if ( dmiFile.exists() &&
+                 dmiFile.open(QIODevice::ReadOnly))
+            {
+                dmiProductName = QString::fromLocal8Bit( dmiFile.readAll().simplified().data() )
+                                 .toLower().replace(dmirx, " ").remove(' ');
+            }
+            if ( !dmiProductName.isEmpty() )
+            {
+                hostnameSuggestion = QString( "%1-%2" ).arg( cleanParts.first() ).arg( dmiProductName );
+            }
+            else
+            {
+               hostnameSuggestion = QString( "%1-pc" ).arg( cleanParts.first() );
+            }
             if ( HOSTNAME_RX.indexIn( hostnameSuggestion ) != -1 )
             {
                 ui->textBoxHostname->setText( hostnameSuggestion );
