diff -up calamares-3.2.28.2/src/modules/locale/Config.cpp.omv~ calamares-3.2.28.2/src/modules/locale/Config.cpp
--- calamares-3.2.28.2/src/modules/locale/Config.cpp.omv~	2020-08-15 18:40:02.935735917 +0200
+++ calamares-3.2.28.2/src/modules/locale/Config.cpp	2020-08-15 18:42:43.213625743 +0200
@@ -62,6 +62,7 @@ loadLocales( const QString& localeGenPat
 
     if ( supported.exists() && supported.open( QIODevice::ReadOnly | QIODevice::Text ) )
     {
+        cDebug() << "Locale: using /usr/share/i18n/SUPPORTED";
         ba = supported.readAll();
         supported.close();
 
@@ -74,16 +75,15 @@ loadLocales( const QString& localeGenPat
     else
     {
         QFile localeGen( localeGenPath );
-        if ( localeGen.open( QIODevice::ReadOnly | QIODevice::Text ) )
+        if ( localeGen.exists() && localeGen.open( QIODevice::ReadOnly | QIODevice::Text ) )
         {
+            cDebug() << "Locale: using " << localeGenPath;
             ba = localeGen.readAll();
             localeGen.close();
         }
         else
         {
-            cWarning() << "Cannot open file" << localeGenPath
-                       << ". Assuming the supported languages are already built into "
-                          "the locale archive.";
+            cDebug() << "Locale: using locale -a output";
             QProcess localeA;
             localeA.start( "locale", QStringList() << "-a" );
             localeA.waitForFinished();
