diff -Naur calamares-3.2.59/src/modules/services-systemd/main.py calamares-3.2.59.tpg/src/modules/services-systemd/main.py
--- calamares-3.2.59/src/modules/services-systemd/main.py	2022-05-29 15:04:24.000000000 +0000
+++ calamares-3.2.59.tpg/src/modules/services-systemd/main.py	2022-06-02 12:04:16.635738137 +0000
@@ -63,14 +63,22 @@
                     description = _("Cannot enable systemd service <code>{name!s}</code>.")
                 elif command == "enable" and suffix == ".target":
                     description = _("Cannot enable systemd target <code>{name!s}</code>.")
+                elif command == "enable" and suffix == ".socket":
+                    description = _("Cannot enable systemd socket <code>{name!s}</code>.")
                 elif command == "enable" and suffix == ".timer":
                     description = _("Cannot enable systemd timer <code>{name!s}</code>.")
                 elif command == "disable" and suffix == ".service":
-                    description = _("Cannot enable systemd service <code>{name!s}</code>.")
+                    description = _("Cannot disable systemd service <code>{name!s}</code>.")
                 elif command == "disable" and suffix == ".target":
                     description = _("Cannot disable systemd target <code>{name!s}</code>.")
+                elif command == "disable" and suffix == ".socket":
+                    description = _("Cannot disable systemd socket <code>{name!s}</code>.")
+                elif command == "disable" and suffix == ".timer":
+                    description = _("Cannot disable systemd timer <code>{name!s}</code>.")
                 elif command == "mask":
                     description = _("Cannot mask systemd unit <code>{name!s}</code>.")
+                elif command == "unmask":
+                    description = _("Cannot unmask systemd unit <code>{name!s}</code>.")
                 else:
                     description = _("Unknown systemd commands <code>{command!s}</code> and <code>{suffix!s}</code> for unit {name!s}.")
 
@@ -103,10 +111,22 @@
     if r is not None:
         return r
 
+    r = systemctl(cfg.get("sockets", []), "enable", ".socket")
+    if r is not None:
+        return r
+
     r = systemctl(cfg.get("disable", []), "disable", ".service")
     if r is not None:
         return r
 
+    r = systemctl(cfg.get("disable-sockets", []), "disable", ".socket")
+    if r is not None:
+        return r
+
+    r = systemctl(cfg.get("disable-timers", []), "disable", ".timer")
+    if r is not None:
+        return r
+
     r = systemctl(cfg.get("disable-targets", []), "disable", ".target")
     if r is not None:
         return r
@@ -115,4 +135,8 @@
     if r is not None:
         return r
 
+    r = systemctl(cfg.get("unmask", []), "unmask", "")
+    if r is not None:
+        return r
+
     return None
