diff -Naur calamares-3.2.32.1/src/modules/partition/gui/PartitionPage.cpp calamares-3.2.32.1.tpg/src/modules/partition/gui/PartitionPage.cpp
--- calamares-3.2.32.1/src/modules/partition/gui/PartitionPage.cpp	2020-10-17 14:41:08.000000000 +0000
+++ calamares-3.2.32.1.tpg/src/modules/partition/gui/PartitionPage.cpp	2020-11-03 19:21:10.976785776 +0000
@@ -170,30 +170,6 @@
         {
             createTable = true;
 
-#ifdef WITH_KPMCORE4API
-            if ( device->type() == Device::Type::SoftwareRAID_Device
-                 && static_cast< SoftwareRAID* >( device )->status() == SoftwareRAID::Status::Inactive )
-            {
-                createTable = false;
-                create = false;
-            }
-#endif
-        }
-        else
-        {
-            currentDeviceIsVG = true;
-
-            LvmDevice* lvmDevice = dynamic_cast< LvmDevice* >( m_core->deviceModel()->deviceForIndex( deviceIndex ) );
-
-            isDeactivable = DeactivateVolumeGroupOperation::isDeactivatable( lvmDevice );
-            isRemovable = RemoveVolumeGroupOperation::isRemovable( lvmDevice );
-
-            isVGdeactivated = m_core->isVGdeactivated( lvmDevice );
-
-            if ( isVGdeactivated )
-            {
-                m_ui->revertButton->setEnabled( true );
-            }
         }
     }
 
@@ -201,9 +177,10 @@
     m_ui->editButton->setEnabled( edit );
     m_ui->deleteButton->setEnabled( del );
     m_ui->newPartitionTableButton->setEnabled( createTable );
-    m_ui->resizeVolumeGroupButton->setEnabled( currentDeviceIsVG && !isVGdeactivated );
-    m_ui->deactivateVolumeGroupButton->setEnabled( currentDeviceIsVG && isDeactivable && !isVGdeactivated );
-    m_ui->removeVolumeGroupButton->setEnabled( currentDeviceIsVG && isRemovable );
+    m_ui->newVolumeGroupButton->setEnabled( false );
+    m_ui->resizeVolumeGroupButton->setEnabled( false );
+    m_ui->deactivateVolumeGroupButton->setEnabled( false );
+    m_ui->removeVolumeGroupButton->setEnabled( false );
 }
 
 void
