From 7a67eed91844eaee0bd4fe2a8fc47c66c182375e Mon Sep 17 00:00:00 2001
From: Teo Mrnjavac <teo@kde.org>
Date: Mon, 5 Dec 2016 17:59:47 +0100
Subject: [PATCH 04/16] Document settings for services module.

---
 src/modules/services/services.conf | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/modules/services/services.conf b/src/modules/services/services.conf
index 50069e6..d9c8895 100644
--- a/src/modules/services/services.conf
+++ b/src/modules/services/services.conf
@@ -15,3 +15,6 @@ targets:
 disable:
   - name: "pacman-init"
     mandatory: false
+
+# Example to express an empty list:
+# disable: []
-- 
2.8.3

