From d7302ca024259cdd3d45b7445f5820c824b42cb4 Mon Sep 17 00:00:00 2001
From: Teo Mrnjavac <teo@kde.org>
Date: Tue, 13 Dec 2016 12:09:36 +0100
Subject: [PATCH 08/16] Provide default for m_writeEtcDefaultKeyboard in
 KeyboardViewStep.

---
 src/modules/keyboard/KeyboardViewStep.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/modules/keyboard/KeyboardViewStep.cpp b/src/modules/keyboard/KeyboardViewStep.cpp
index e6521e1..cdb2657 100644
--- a/src/modules/keyboard/KeyboardViewStep.cpp
+++ b/src/modules/keyboard/KeyboardViewStep.cpp
@@ -29,6 +29,7 @@ KeyboardViewStep::KeyboardViewStep( QObject* parent )
     : Calamares::ViewStep( parent )
     , m_widget( new KeyboardPage() )
     , m_nextEnabled( false )
+    , m_writeEtcDefaultKeyboard( true )
 {
     m_widget->init();
     m_nextEnabled = true;
-- 
2.8.3

