From 89f078f8c5d83dfd34e2ed855d8d23a9aed9fd63 Mon Sep 17 00:00:00 2001
From: Teo Mrnjavac <teo@kde.org>
Date: Thu, 22 Dec 2016 13:52:33 +0100
Subject: [PATCH 13/16] Alignment.

---
 src/modules/webview/WebViewStep.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/modules/webview/WebViewStep.cpp b/src/modules/webview/WebViewStep.cpp
index 76bee9a..069b52d 100644
--- a/src/modules/webview/WebViewStep.cpp
+++ b/src/modules/webview/WebViewStep.cpp
@@ -34,9 +34,9 @@ WebViewStep::WebViewStep( QObject* parent )
     : Calamares::ViewStep( parent )
 {
     emit nextStatusChanged( true );
-    #ifdef WEBVIEW_WITH_WEBENGINE
-        QtWebEngine::initialize();
-    #endif
+#ifdef WEBVIEW_WITH_WEBENGINE
+    QtWebEngine::initialize();
+#endif
     m_view = new C_QWEBVIEW();
 #ifdef WEBVIEW_WITH_WEBKIT
     m_view->settings()->setFontFamily( QWebSettings::StandardFont,
-- 
2.8.3

