From 5f231086de09dda1473d94dc091031166e26d38f Mon Sep 17 00:00:00 2001
From: Chantara Tith <tith.chantara@gmail.com>
Date: Mon, 12 Sep 2016 18:25:25 +0700
Subject: [PATCH] Add LVM devices to devies-list.

---
 src/modules/partition/core/PartitionCoreModule.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/modules/partition/core/PartitionCoreModule.cpp b/src/modules/partition/core/PartitionCoreModule.cpp
index 7733384..6356348 100644
--- a/src/modules/partition/core/PartitionCoreModule.cpp
+++ b/src/modules/partition/core/PartitionCoreModule.cpp
@@ -42,6 +42,7 @@
 
 // KPMcore
 #include <kpmcore/core/device.h>
+#include <kpmcore/core/lvmdevice.h>
 #include <kpmcore/core/partition.h>
 #include <kpmcore/backend/corebackend.h>
 #include <kpmcore/backend/corebackendmanager.h>
@@ -119,6 +120,8 @@ PartitionCoreModule::doInit()
 
     using DeviceList = QList< Device* >;
     DeviceList devices = PartUtils::getDevices( PartUtils::DeviceType::WritableOnly );
+    for ( auto dev : LvmDevice::scanSystemLVM() )
+        devices.append(dev);
 
     cDebug() << "LIST OF DETECTED DEVICES:";
     cDebug() << "node\tcapacity\tname\tprettyName";
