--- calamares-3.1.1/src/modules/welcome/checker/partman_devices.c~	2017-06-23 09:43:24.000000000 +0100
+++ calamares-3.1.1/src/modules/welcome/checker/partman_devices.c	2017-09-04 10:02:05.050543203 +0100
@@ -126,8 +126,14 @@
             break;
         }
     }
-    ped_device_free_all();
-
+   // We would free the devices to release allocated memory,
+    // but other modules might be using partman use well,
+    // and they can hold pointers to libparted structures in
+    // other threads.
+    //
+    // So prefer to leak memory, instead.
+    //
+    // ped_device_free_all();
     return big_enough;
 }
 
