diff -Naur calamares-0.17.0-20150112/settings.conf calamares-0.17.0-20150112.tpg/settings.conf
--- calamares-0.17.0-20150112/settings.conf	2015-01-11 13:53:23.000000000 +0000
+++ calamares-0.17.0-20150112.tpg/settings.conf	2015-01-12 21:59:55.719975775 +0000
@@ -4,7 +4,7 @@
 # Modules can be job modules (with different interfaces) and QtWidgets view modules.
 # They could all be placed in a number of different paths.
 # "local" is LIBDIR/calamares/modules with settings in SHARE/calamares/modules
-modules-search: [ local, /path/to/dir/with/more/modules ]
+modules-search: [ local ]
 
 # We define the module names in the order they should show up (QtWidget view modules,
 # with one or more pages) OR be executed if enqueued (all other modules).
@@ -39,18 +39,17 @@
 - locale
 - keyboard
 - localecfg
-- initcpiocfg
-- initcpio
 - users
-#- displaymanager
+- displaymanager
 - networkcfg
 - hwclock
 - services
-#- dracut
+- dracut
 #- initramfs
 - grubcfg
 - grub
 #- bootloader
+- packages
 - umount
 
 # Phase 3 - postinstall.
@@ -67,4 +66,4 @@
 # paths relative to the current directory.
 # Only the name of the branding component (directory) should be specified here, Calamares
 # then takes care of finding it and loading the contents.
-branding: default
+branding: auto
diff -Naur calamares-0.17.0-20150112/src/modules/grub/grub.conf calamares-0.17.0-20150112.tpg/src/modules/grub/grub.conf
--- calamares-0.17.0-20150112/src/modules/grub/grub.conf	2015-01-11 13:53:23.000000000 +0000
+++ calamares-0.17.0-20150112.tpg/src/modules/grub/grub.conf	2015-01-12 22:02:04.588501788 +0000
@@ -1,11 +1,11 @@
 ---
 # GRUB 2 binary names and boot directory
 # Some distributions (e.g. Fedora) use grub2-* (resp. /boot/grub2/) names.
-grubInstall: "grub-install"
-grubMkconfig: "grub-mkconfig"
-grubCfg: "/boot/grub/grub.cfg"
+grubInstall: "grub2-install"
+grubMkconfig: "grub2-mkconfig"
+grubCfg: "/boot/grub2/grub.cfg"
 # Optionally set the --bootloader-id to use for EFI. If not set, this defaults
 # to the bootloaderEntryName from branding.desc with problematic characters
 # replaced. If an efiBootloaderId is specified here, it is taken to already be a
 # valid directory name, so no such postprocessing is done in this case.
-# efiBootloaderId: "dirname"
+efiBootloaderId: "openmandriva"
diff -Naur calamares-0.17.0-20150112/src/modules/packages/packages.conf calamares-0.17.0-20150112.tpg/src/modules/packages/packages.conf
--- calamares-0.17.0-20150112/src/modules/packages/packages.conf	2015-01-11 13:53:23.000000000 +0000
+++ calamares-0.17.0-20150112.tpg/src/modules/packages/packages.conf	2015-01-12 22:03:33.478944661 +0000
@@ -9,7 +9,7 @@
 #  - apt         - APT frontend for DEB and RPM
 #  - pacman      - Pacman
 #
-backend: packagekit
+backend: urpmi
 #
 # List of maps with package operations such as install or remove.
 # Distro developers can provide a list of packages to remove
@@ -24,18 +24,6 @@
 # storage called "packageOperations" and it is processed
 # after the static list in the job configuration.
 #
-#operations:
-#  - install:
-#      - pkg1
-#      - pkg2
-#  - remove:
-#      - pkg3
-#      - pkg4
-#  - install:
-#      - pkg5
-#  - remove:
-#      - pkg2
-#      - pkg1
-#    install:
-#      - pkgs6
-#      - pkg7
+operations:
+  - remove:
+    - calamares
diff -Naur calamares-0.17.0-20150112/src/modules/unpackfs/unpackfs.conf calamares-0.17.0-20150112.tpg/src/modules/unpackfs/unpackfs.conf
--- calamares-0.17.0-20150112/src/modules/unpackfs/unpackfs.conf	2015-01-11 13:53:23.000000000 +0000
+++ calamares-0.17.0-20150112.tpg/src/modules/unpackfs/unpackfs.conf	2015-01-12 22:06:36.008954515 +0000
@@ -1,8 +1,5 @@
 ---
 unpack:
-    -   source: "/path/to/filesystem.img"
-        sourcefs: "ext4"
-        destination: ""
-    -   source: "/path/to/another/filesystem.sqfs"
+    -   source: "/live/media/LiveOS/squashfs.img"
         sourcefs: "squashfs"
         destination: ""
