diff -Naur calamares-1.0.0-20150203/src/modules/packages/main.py calamares-1.0.0-20150203.tpg/src/modules/packages/main.py
--- calamares-1.0.0-20150203/src/modules/packages/main.py	2015-02-03 08:45:42.000000000 +0000
+++ calamares-1.0.0-20150203.tpg/src/modules/packages/main.py	2015-02-22 00:35:34.360391343 +0000
@@ -53,7 +53,7 @@
             # ignore the error code for now because dnf thinks removing a nonexistent package is an error
             chroot_call(["dnf", "--disablerepo=*", "-C", "-y", "remove"] + pkgs)
         elif self.backend == "urpmi":
-            check_chroot_call(["urpme"] + pkgs)
+            check_chroot_call(["urpme", "--auto"] + pkgs)
         elif self.backend == "apt":
             check_chroot_call(["apt-get", "--purge", "-q", "-y", "remove"] + pkgs)
             check_chroot_call(["apt-get", "--purge", "-q", "-y", "autoremove"])
