diff -Naur calamares-2.2.3/calamares.desktop calamares-2.2.3.tpg/calamares.desktop
--- calamares-2.2.3/calamares.desktop	2016-05-27 15:22:15.000000000 +0000
+++ calamares-2.2.3.tpg/calamares.desktop	2016-05-31 18:55:21.612580822 +0000
@@ -1,12 +1,11 @@
 [Desktop Entry]
 Type=Application
 Version=1.0
-Name=Calamares
-GenericName=System Installer
+Name=Install OpenMandriva Lx
 TryExec=calamares
 Exec=pkexec /usr/bin/calamares
-Comment=Calamares — System Installer
-Icon=calamares
+Comment=OpenMandriva Lx installer
+Icon=/usr/share/icons/openmandriva-install.svg
 Terminal=false
 StartupNotify=true
 Categories=Qt;System;
