diff -up calamares-3.2.2-20190106/src/modules/partition/core/PartitionCoreModule.cpp.omv~ calamares-3.2.2-20190106/src/modules/partition/core/PartitionCoreModule.cpp
--- calamares-3.2.2-20190106/src/modules/partition/core/PartitionCoreModule.cpp.omv~	2019-05-05 17:42:49.881958480 +0200
+++ calamares-3.2.2-20190106/src/modules/partition/core/PartitionCoreModule.cpp	2019-05-05 17:45:35.051306625 +0200
@@ -672,7 +672,7 @@ PartitionCoreModule::scanForLVMPVs()
     }
 
     // Update LVM::pvList
-    LvmDevice::scanSystemLVM( physicalDevices );
+    // [should happen automatically with kpmcore 4.0] LvmDevice::scanSystemLVM( physicalDevices );
 
 #ifdef WITH_KPMCOREGT33
     for ( auto p : LVM::pvList::list() )
