diff -Naur calamares-3.0/calamares.desktop calamares-3.0.tpg/calamares.desktop
--- calamares-3.0/calamares.desktop	2017-01-20 15:18:23.000000000 +0000
+++ calamares-3.0.tpg/calamares.desktop	2017-01-23 20:07:09.073333708 +0000
@@ -1,13 +1,12 @@
 [Desktop Entry]
 Type=Application
 Version=1.0
-Name=Calamares
-GenericName=System Installer
+Name=Install OpenMandriva Lx
 Keywords=calamares;system;installer
 TryExec=calamares
 Exec=pkexec /usr/bin/calamares
-Comment=Calamares — System Installer
-Icon=calamares
+Comment=OpenMandriva Lx installer
+Icon=/usr/share/icons/openmandriva-install.svg
 Terminal=false
 StartupNotify=true
 Categories=Qt;System;
