diff -up calamares-3.2.12/calamares.desktop.1~ calamares-3.2.12/calamares.desktop
--- calamares-3.2.12/calamares.desktop.1~	2019-08-22 15:28:36.975064504 +0200
+++ calamares-3.2.12/calamares.desktop	2019-08-22 15:30:12.972355573 +0200
@@ -1,13 +1,13 @@
 [Desktop Entry]
 Type=Application
 Version=1.0
-Name=Install System
-GenericName=System Installer
+Name=Install OpenMandriva Lx
+GenericName=OpenMandriva Installer
 Keywords=calamares;system;installer;
 TryExec=calamares
 Exec=pkexec /usr/bin/calamares
-Comment=Calamares — System Installer
-Icon=calamares
+Comment=OpenMandriva Lx Installer
+Icon=openmandriva-install
 Terminal=false
 StartupNotify=true
 Categories=Qt;System;
diff -up calamares-3.2.12/calamares.desktop.in.1~ calamares-3.2.12/calamares.desktop.in
--- calamares-3.2.12/calamares.desktop.in.1~	2019-08-22 15:28:36.975064504 +0200
+++ calamares-3.2.12/calamares.desktop.in	2019-08-22 15:30:55.552484597 +0200
@@ -1,13 +1,13 @@
 [Desktop Entry]
 Type=Application
 Version=1.0
-Name=Install System
-GenericName=System Installer
+Name=Install OpenMandriva
+GenericName=OpenMandriva Lx Installer
 Keywords=calamares;system;installer;
 TryExec=calamares
 Exec=pkexec /usr/bin/calamares
-Comment=Calamares — System Installer
-Icon=calamares
+Comment=OpenMandriva Installer
+Icon=openmandriva-install
 Terminal=false
 StartupNotify=true
 Categories=Qt;System;
