diff -Naur calamares.org/src/modules/removeuser/main.py calamares.removeuser/src/modules/removeuser/main.py
--- calamares.org/src/modules/removeuser/main.py	1970-01-01 00:00:00.000000000 +0000
+++ calamares.removeuser/src/modules/removeuser/main.py	2015-02-25 21:18:01.777350068 +0000
@@ -0,0 +1,35 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# === This file is part of Calamares - <http://github.com/calamares> ===
+#
+#   Copyright 2015, Teo Mrnjavac <teo@kde.org>
+#
+#   Calamares is free software: you can redistribute it and/or modify
+#   it under the terms of the GNU General Public License as published by
+#   the Free Software Foundation, either version 3 of the License, or
+#   (at your option) any later version.
+#
+#   Calamares is distributed in the hope that it will be useful,
+#   but WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+#   GNU General Public License for more details.
+#
+#   You should have received a copy of the GNU General Public License
+#   along with Calamares. If not, see <http://www.gnu.org/licenses/>.
+
+import subprocess
+import libcalamares
+
+
+def run():
+    """ Remove live user from target system """
+
+    username = libcalamares.job.configuration[("username")]
+    try:
+        libcalamares.utils.check_chroot_call(["userdel", "-f", "-r", username])
+    except subprocess.CalledProcessError as e:
+        libcalamares.utils.debug( "Cannot remove user. " +
+                                  "userdel terminated with exit code {}.".format(e.returncode))
+
+    return None
diff -Naur calamares.org/src/modules/removeuser/module.desc calamares.removeuser/src/modules/removeuser/module.desc
--- calamares.org/src/modules/removeuser/module.desc	1970-01-01 00:00:00.000000000 +0000
+++ calamares.removeuser/src/modules/removeuser/module.desc	2015-02-25 21:18:01.777350068 +0000
@@ -0,0 +1,6 @@
+---
+type:       "job"
+name:       "removeuser"
+interface:  "python"
+requires:   []
+script:     "main.py"
diff -Naur calamares.org/src/modules/removeuser/removeuser.conf calamares.removeuser/src/modules/removeuser/removeuser.conf
--- calamares.org/src/modules/removeuser/removeuser.conf	1970-01-01 00:00:00.000000000 +0000
+++ calamares.removeuser/src/modules/removeuser/removeuser.conf	2015-02-25 21:18:01.777350068 +0000
@@ -0,0 +1,2 @@
+---
+username: live
