diff -Naur calamares-0.17.0-20150112/src/modules/packages/main.py calamares-0.17.0-20150112.tpg/src/modules/packages/main.py
--- calamares-0.17.0-20150112/src/modules/packages/main.py	2015-01-11 13:53:23.000000000 +0000
+++ calamares-0.17.0-20150112.tpg/src/modules/packages/main.py	2015-01-16 21:31:27.425756990 +0000
@@ -35,7 +35,7 @@
         elif self.backend == "dnf":
             check_chroot_call(["dnf", "install", "-y"] + pkgs)
         elif self.backend == "urpmi":
-            check_chroot_call(["urpmi"] + pkgs)
+            check_chroot_call(["urpmi", "--download-all", "--no-suggests", "--no-verify-rpm", "--fastunsafe", "--ignoresize", "--nolock", "--auto"] + pkgs)
         elif self.backend == "apt":
             check_chroot_call(["apt-get", "-q", "-y", "install"] + pkgs)
         elif self.backend == "pacman":
