diff -Naur calamares-3.2.20/src/modules/machineid/MachineIdJob.cpp calamares-3.2.20.tpg/src/modules/machineid/MachineIdJob.cpp
--- calamares-3.2.20/src/modules/machineid/MachineIdJob.cpp	2020-02-27 22:26:29.000000000 +0000
+++ calamares-3.2.20.tpg/src/modules/machineid/MachineIdJob.cpp	2020-03-12 10:32:28.661044992 +0000
@@ -66,7 +66,7 @@
 
     QString target_systemd_machineid_file = QStringLiteral( "/etc/machine-id" );
     QString target_dbus_machineid_file = QStringLiteral( "/var/lib/dbus/machine-id" );
-    QString target_entropy_file = QStringLiteral( "/var/lib/urandom/random-seed" );
+    QString target_entropy_file = QStringLiteral( "/var/lib/systemd/random-seed" );
 
     const CalamaresUtils::System* system = CalamaresUtils::System::instance();
 
