diff -Naur calamares-3.2.7/src/modules/partition/gui/PartitionViewStep.cpp calamares-3.2.7-p/src/modules/partition/gui/PartitionViewStep.cpp
--- calamares-3.2.7/src/modules/partition/gui/PartitionViewStep.cpp	2019-04-27 20:12:03.000000000 +0200
+++ calamares-3.2.7-p/src/modules/partition/gui/PartitionViewStep.cpp	2019-06-12 15:26:42.156998632 +0200
@@ -405,7 +405,7 @@
                                   "<br/><br/>"
                                   "To configure an EFI system partition, go back and "
                                   "select or create a FAT32 filesystem with the "
-                                  "<strong>esp</strong> flag enabled and mount point "
+                                  "<strong>boot</strong> flag enabled and mount point "
                                   "<strong>%2</strong>.<br/><br/>"
                                   "You can continue without setting up an EFI system "
                                   "partition but your system may fail to start." )
@@ -418,7 +418,7 @@
                 description = tr( "An EFI system partition is necessary to start %1."
                                   "<br/><br/>"
                                   "A partition was configured with mount point "
-                                  "<strong>%2</strong> but its <strong>esp</strong> "
+                                  "<strong>%2</strong> but its <strong>boot</strong> "
                                   "flag is not set.<br/>"
                                   "To set the flag, go back and edit the partition."
                                   "<br/><br/>"
