diff -Naur calamares-3.2.16/src/modules/fsresizer/CMakeLists.txt calamares-3.2.16.tpg/src/modules/fsresizer/CMakeLists.txt
--- calamares-3.2.16/src/modules/fsresizer/CMakeLists.txt	2019-11-01 16:24:10.000000000 +0000
+++ calamares-3.2.16.tpg/src/modules/fsresizer/CMakeLists.txt	2019-11-05 09:52:59.830005159 +0000
@@ -1,4 +1,4 @@
-find_package( KPMcore 3.3 )
+find_package( KPMcore )
 find_package( KF5Config CONFIG )
 find_package( KF5I18n CONFIG )
 find_package( KF5WidgetsAddons CONFIG )
@@ -8,10 +8,10 @@
 if ( KPMcore_FOUND AND Qt5DBus_FOUND AND KF5CoreAddons_FOUND AND KF5Config_FOUND )
     include_directories( ${KPMCORE_INCLUDE_DIR} ${CMAKE_SOURCE_DIR}/src/modules/partition )
 
-    if ( KPMcore_VERSION VERSION_GREATER "3.3.0")
+    if ( KPMcore_VERSION VERSION_GREATER "3.3.0"  AND KPMcore_VERSION LESS "3.80.0")
         list( APPEND _partition_defs WITH_KPMCORE331API) # kpmcore > 3.3.0 with deprecations
     endif()
-    if ( KPMcore_VERSION VERSION_GREATER "3.90")
+    if ( KPMcore_VERSION VERSION_GREATER "3.80.0")
         list( APPEND _partition_defs WITH_KPMCORE4API) # kpmcore 4 with new API
     endif()
 
diff -Naur calamares-3.2.16/src/modules/partition/CMakeLists.txt calamares-3.2.16.tpg/src/modules/partition/CMakeLists.txt
--- calamares-3.2.16/src/modules/partition/CMakeLists.txt	2019-11-01 16:24:10.000000000 +0000
+++ calamares-3.2.16.tpg/src/modules/partition/CMakeLists.txt	2019-11-05 09:54:10.166005574 +0000
@@ -17,7 +17,7 @@
 
 find_package(ECM ${ECM_VERSION} REQUIRED NO_MODULE)
 
-find_package( KPMcore 3.3 )
+find_package( KPMcore )
 set_package_properties(
     KPMcore PROPERTIES
     PURPOSE "For partitioning module"
@@ -27,10 +27,10 @@
 find_package( KF5WidgetsAddons CONFIG )
 
 if ( KPMcore_FOUND AND Qt5DBus_FOUND AND KF5CoreAddons_FOUND AND KF5Config_FOUND )
-    if ( KPMcore_VERSION VERSION_GREATER "3.3.0")
+    if ( KPMcore_VERSION VERSION_GREATER "3.3.0"  AND KPMcore_VERSION LESS "3.80.0")
         list( APPEND _partition_defs WITH_KPMCORE331API) # kpmcore > 3.3.0 with deprecations
     endif()
-    if ( KPMcore_VERSION VERSION_GREATER "3.90")
+    if ( KPMcore_VERSION VERSION_GREATER "3.80.0")
         list( APPEND _partition_defs WITH_KPMCORE4API) # kpmcore 4 with new API
     endif()
 
diff -Naur calamares-3.2.16/src/modules/partition/core/KPMHelpers.h calamares-3.2.16.tpg/src/modules/partition/core/KPMHelpers.h
--- calamares-3.2.16/src/modules/partition/core/KPMHelpers.h	2019-11-01 16:24:10.000000000 +0000
+++ calamares-3.2.16.tpg/src/modules/partition/core/KPMHelpers.h	2019-11-05 09:55:37.184006087 +0000
@@ -34,7 +34,7 @@
 class PartitionNode;
 class PartitionRole;
 
-#ifdef WITH_KPMCORE331API
+#ifdef WITH_KPMCORE4API
 #define KPM_PARTITION_FLAG(x) PartitionTable::Flag::x
 #define KPM_PARTITION_STATE(x) Partition::State::x
 #define KPM_PARTITION_FLAG_ESP PartitionTable::Flag::Boot
diff -Naur calamares-3.2.16/src/modules/partition/core/PartitionCoreModule.cpp calamares-3.2.16.tpg/src/modules/partition/core/PartitionCoreModule.cpp
--- calamares-3.2.16/src/modules/partition/core/PartitionCoreModule.cpp	2019-11-01 16:24:10.000000000 +0000
+++ calamares-3.2.16.tpg/src/modules/partition/core/PartitionCoreModule.cpp	2019-11-05 09:56:40.758006462 +0000
@@ -857,8 +857,9 @@
     {
         if ( part->mountPoint() == "/" )
         {
+            // use KPM_PARTITION_FLAG_ESP so it works with both APIs of kpmcore
             createPartition( dev, part,
-                             part->activeFlags() | ( isEfi ? KPM_PARTITION_FLAG(None) : KPM_PARTITION_FLAG(Boot) )
+                             part->activeFlags() | ( isEfi ? KPM_PARTITION_FLAG(None) : KPM_PARTITION_FLAG_ESP )
                            );
         }
         else
