diff -Naur calamares-3.2.2/src/modules/services-systemd/main.py calamares-3.2.2-p/src/modules/services-systemd/main.py
--- calamares-3.2.2/src/modules/services-systemd/main.py	2018-09-04 14:03:59.000000000 +0200
+++ calamares-3.2.2-p/src/modules/services-systemd/main.py	2019-01-04 19:40:52.723729166 +0100
@@ -92,6 +92,10 @@
     if r is not None:
         return r
 
+    r = systemctl(cfg.get("unmask", []), "unmask", "")
+    if r is not None:
+        return r
+
 
     # This could have just been return r
     return None
diff -Naur calamares-3.2.2/src/modules/services-systemd/services-systemd.conf calamares-3.2.2-p/src/modules/services-systemd/services-systemd.conf
--- calamares-3.2.2/src/modules/services-systemd/services-systemd.conf	2018-09-04 14:03:59.000000000 +0200
+++ calamares-3.2.2-p/src/modules/services-systemd/services-systemd.conf	2019-01-04 19:43:54.506147190 +0100
@@ -61,6 +61,9 @@
 # mask:
 #  - name: "NetworkManager.service"
 #  - mandatory: true
+# unmask:
+#  - name: "some.service"
+#  - mandatory: tru
 
 # By default, no changes are made.
 services: []
@@ -68,3 +71,4 @@
 disable: []
 disable-targets: []
 mask: []
+unmask: []
